onchain/rollups/contracts/consensus/quorum/Quorum.sol

```diff
@@ -69,25 +82,99 @@ contract Quorum is AbstractConsensus, PaymentSplitter {
-        uint256 index = validatorIndex[msg.sender];
-        if (index == 0) {
-            revert OnlyValidator();
-        }
+        uint256 id = validatorId[msg.sender];
+        require(id != 0, "Quorum: sender is not validator");
 
-        // If the msg.sender hasn't submitted the same claim before
-        Votes storage claimVotes = votes[_claimData];
-        if (!claimVotes.votedBitMap.get(index)) {
-            claimVotes.votedBitMap.set(index);
+        VotingStatus storage votingStatus = votingStatuses[_claimData];
+        BitMaps.BitMap storage inFavorById = votingStatus.inFavorById;
+
+        if (!inFavorById.get(id)) {
+            // If validator hasn't voted yet, cast their vote
+            inFavorById.set(id);
 
             // If this claim has now just over half of the quorum's votes,
             // then we can submit it to the history contract.
-            if (++claimVotes.count == 1 + numOfValidators / 2) {
+            if (++votingStatus.inFavorCount == 1 + numOfValidators / 2) {
                 history.submitClaim(_claimData);
             }
         }
     }
 
+    /// @notice Get an array with the IDs of all validators.
+    /// @return Array of IDs of validators
+    function validatorIds() external view returns (uint256[] memory) {
+        uint256[] memory array = new uint256[](numOfValidators);
+
+        uint256 id = 1;
+        for (uint256 i; i < numOfValidators; ++i) {
+            array[i] = id;
+            ++id;
+        }
+
+        return array;
+    }
+
+    /// @notice Get an array with the addresses of all validators.
+    /// @return Array of addresses of validators
+    function validators() external view returns (address[] memory) {
+        address[] memory array = new address[](numOfValidators);
+
+        uint256 id = 1;
+        for (uint256 i; i < numOfValidators; ++i) {
+            array[i] = validatorById[id];
+            ++id;
+        }
+
+        return array;
+    }
+
+    /// @notice Get the number of validator in favor of a claim.
+    /// @param _claimData Data for submitting a claim
+    /// @return Number of validator in favor of claim.
+    function numOfValidatorsInFavorOf(
+        bytes calldata _claimData
+    ) external view returns (uint256) {
+        VotingStatus storage votingStatus = votingStatuses[_claimData];
+        return votingStatus.inFavorCount;
+    }
+
+    /// @notice Check whether a validator is in favor of a claim.
+    /// @param _validatorId The ID of the validator
+    /// @param _claimData Data for submitting a claim
+    /// @return Array of addresses of validators in favor of claim
+    /// @dev Assumes the provided ID is valid
+    function isValidatorInFavorOf(
+        uint256 _validatorId,
+        bytes calldata _claimData
+    ) external view returns (bool) {
+        VotingStatus storage votingStatus = votingStatuses[_claimData];
+        BitMaps.BitMap storage inFavorById = votingStatus.inFavorById;
```

Comment:
Should we make these 2 lines into 1?
```
BitMaps.BitMap storage inFavorById = votingStatuses[_claimData].inFavorById;
```

Reply:
We could, but then `prettier` would break the line in a not-so-pretty way.

Reply:
I see. Do you think it would be more readable even though broken by prettier? Or it doesn't make that much of a difference

Reply:
I think it's more readable the way it currently is now, broken in two lines. :-)